python/sdist/amici/pysb_import.py

```diff
@@ -178,6 +178,10 @@ def pysb2amici(
         compiler=compiler,
         generate_sensitivity_code=generate_sensitivity_code,
     )
+    # Sympy code optimizations are incompatible with PySB objects, as
+    #  `pysb.Observable` comes with its own `.match` which overrides
+    #  `sympy.Basic.match()`, breaking `sympy.codegen.rewriting.optimize`.
+    exporter._code_printer._fpoptimizer = None
```

Comment:
:( sounds like this is something we want to get fixed. We could convert everything to expressions.

Reply:
Ideally there wouldn't be any PySB objects anymore at the stage of code generation.